lit_nlp/examples/datasets/question_answering.py

```diff
@@ -0,0 +1,31 @@
+"""Data loaders for summarization datasets."""
+
+from lit_nlp.api import dataset as lit_dataset
+from lit_nlp.api import types as lit_types
+import tensorflow_datasets as tfds
+
+
+class TyDiQA(lit_dataset.Dataset):
+  """TyDiQA dataset."""
+
+  def __init__(self, split="validation-en", max_examples=-1, max_seq_len=500):
+    """Dataset constructor, loads the data into memory."""
+    ds = tfds.load("tydi_qa", split=split)
+
+    # into datafrane to decode string
+    df = tfds.as_dataframe(ds.take(max_examples))
+    df['context'] = df['context'].str.decode("utf-8")
+    df['question'] = df['question'].str.decode("utf-8")
+
+    # populate this with data records
+    self._examples = [{
+      'context': row['context'],
+      'question': row['question'],
+    } for _, row in df.iterrows()]
```

Comment:
Given the impending changes to the `spec()` for this Dataset, we'll need to update this comprehension. Unfortunately, adding the other fields will violate the [Style Guide's rules for comprehensions](https://google.github.io/styleguide/pyguide.html#27-comprehensions--generator-expressions) (i.e., the mapping statement won't fit on one line), so we'll need to switch this to a for loop. 

```suggestion
    self._examples = []
    for _, row in df.iterrows():
      self._examples.append({
          # Field mapping code goes here
      })
```